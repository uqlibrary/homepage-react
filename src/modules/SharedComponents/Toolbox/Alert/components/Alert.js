import React, { useState } from 'react';
import { PropTypes } from 'prop-types';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import CircularProgress from '@material-ui/core/CircularProgress';
import Close from '@material-ui/icons/Close';

import ErrorOutline from '@material-ui/icons/ErrorOutline';
import Error from '@material-ui/icons/Error';
import Warning from '@material-ui/icons/Warning';
import Info from '@material-ui/icons/Info';
import InfoOutlined from '@material-ui/icons/InfoOutlined';
import Help from '@material-ui/icons/Help';
import HelpOutline from '@material-ui/icons/HelpOutline';
import Done from '@material-ui/icons/Done';
import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';
import { makeStyles } from '@material-ui/styles';

const classNames = require('classnames');

const useStyles = makeStyles(
    theme => ({
        common: {
            borderRadius: 0,
            padding: '8px 37px',
        },
        wrapper: {
            [theme.breakpoints.up('xs')]: {
                maxWidth: 1200,
                marginLeft: 'auto',
                marginRight: 'auto',
                paddingRight: 0,
                paddingLeft: 0,
            },
            [theme.breakpoints.down('xs')]: {
                maxWidth: '100%',
                marginLeft: 'auto',
                marginRight: 'auto',
                paddingRight: 0,
                paddingLeft: 0,
            },
        },
        '@keyframes wiggle': {
            from: { transform: 'rotate(-30deg)', transformOrigin: '40% 50%' },
            to: { transform: 'rotate(15deg)', transformOrigin: '40% 50%' },
        },
        wiggler: {
            animationName: '$wiggle',
            animationDuration: '0.3s',
            animationIterationCount: 20,
            animationDirection: 'alternate',
            animationTimingFunction: 'ease-in-out',
        },
        icon: {
            '& .icon': {
                fontSize: 28,
                marginRight: 12,
                marginBottom: -6,
                marginLeft: -6,
            },
            '& .spinner': {
                margin: '8px 24px 0 6px',
            },
        },
        text: {
            alignSelf: 'center',
            padding: '4px 0',
            textShadow: '1px 1px 1px rgba(0, 0, 0, 0.2)',
            verticalAlign: 'middle',
        },
        actionButton: {
            marginRight: -4,
            '& .action': {
                [theme.breakpoints.down('xs')]: {
                    marginRight: 12,
                },
            },
        },
        dismissButton: {
            marginLeft: -8,
            marginRight: -19,
            marginTop: -12,
            marginBottom: -12,
        },
        linked: {
            '&:hover': {
                cursor: 'pointer',
            },
        },
        error: {
            color: theme.palette.white.main,
            backgroundColor: theme.palette.error.main,
            '& a:link, & a:hover, & a:visited': {
                color: theme.palette.white.main,
                textDecoration: 'underline',
            },
            '& .spinner': {
                color: theme.palette.error.dark,
            },
            '& .icon': {
                color: theme.palette.error.dark,
            },
            '& .dismiss': {
                color: theme.palette.error.dark,
            },
            '& .action': {
                color: theme.palette.white.main,
                backgroundColor: theme.palette.error.dark,
            },
        },
        error_outline: {
            color: theme.palette.white.main,
            backgroundColor: theme.palette.error.main,
            '& a:link, & a:hover, & a:visited': {
                color: theme.palette.white.main,
                textDecoration: 'underline',
            },
            '& .spinner': {
                color: theme.palette.error.dark,
            },
            '& .icon': {
                color: theme.palette.error.dark,
            },
            '& .dismiss': {
                color: theme.palette.error.dark,
            },
            '& .action': {
                color: theme.palette.white.main,
                backgroundColor: theme.palette.error.dark,
            },
        },
        warning: {
            color: theme.palette.white.main,
            backgroundColor: theme.palette.warning.main,
            '& a:link, & a:hover, & a:visited': {
                color: theme.palette.white.main,
                textDecoration: 'underline',
            },
            '& .spinner': {
                color: theme.palette.warning.dark,
            },
            '& .icon': {
                color: theme.palette.warning.dark,
            },
            '& .dismiss': {
                color: theme.palette.warning.dark,
            },
            '& .action': {
                color: theme.palette.white.main,
                backgroundColor: theme.palette.warning.dark,
            },
        },
        help: {
            color: theme.palette.white.main,
            backgroundColor: theme.palette.secondary.main,
            '& a:link, & a:hover, & a:visited': {
                color: theme.palette.white.main,
                textDecoration: 'underline',
            },
            '& .spinner': {
                color: theme.palette.secondary.dark,
            },
            '& .icon': {
                color: theme.palette.secondary.dark,
            },
            '& .dismiss': {
                color: theme.palette.secondary.dark,
            },
            '& .action': {
                color: theme.palette.white.main,
                backgroundColor: theme.palette.secondary.dark,
            },
        },
        help_outline: {
            color: theme.palette.white.main,
            backgroundColor: theme.palette.secondary.main,
            '& a:link, & a:hover, & a:visited': {
                color: theme.palette.white.main,
                textDecoration: 'underline',
            },
            '& .spinner': {
                color: theme.palette.secondary.dark,
            },
            '& .icon': {
                color: theme.palette.secondary.dark,
            },
            '& .dismiss': {
                color: theme.palette.secondary.dark,
            },
            '& .action': {
                color: theme.palette.white.main,
                backgroundColor: theme.palette.secondary.dark,
            },
        },
        info: {
            color: theme.palette.white.main,
            backgroundColor: theme.palette.accent.main,
            '& a:link, & a:hover, & a:visited': {
                color: theme.palette.white.main,
                textDecoration: 'underline',
            },
            '& .spinner': {
                color: theme.palette.accent.dark,
            },
            '& .icon': {
                color: theme.palette.accent.dark,
            },
            '& .dismiss': {
                color: theme.palette.accent.dark,
            },
            '& .action': {
                color: theme.palette.white.main,
                backgroundColor: theme.palette.accent.dark,
            },
        },
        info_outline: {
            color: theme.palette.white.main,
            backgroundColor: theme.palette.accent.main,
            '& a:link, & a:hover, & a:visited': {
                color: theme.palette.white.main,
                textDecoration: 'underline',
            },
            '& .spinner': {
                color: theme.palette.accent.dark,
            },
            '& .icon': {
                color: theme.palette.accent.dark,
            },
            '& .dismiss': {
                color: theme.palette.accent.dark,
            },
            '& .action': {
                color: theme.palette.white.main,
                backgroundColor: theme.palette.accent.dark,
            },
        },
        done: {
            color: theme.palette.white.main,
            backgroundColor: theme.palette.success.light,
            '& a:link, & a:hover, & a:visited': {
                color: theme.palette.white.main,
                textDecoration: 'underline',
            },
            '& .spinner': {
                color: theme.palette.success.dark,
            },
            '& .icon': {
                color: theme.palette.success.dark,
            },
            '& .dismiss': {
                color: theme.palette.success.dark,
            },
            '& .action': {
                color: theme.palette.white.main,
                backgroundColor: theme.palette.success.dark,
            },
        },
    }),
    { withTheme: true },
);

export const Alert = ({
    action,
    actionButtonLabel,
    allowDismiss,
    customIcon,
    customType,
    disableAlertClick,
    dismissAction,
    dismissTitle,
    message,
    showLoader,
    alertId,
    title,
    type,
    wiggle,
    canHide,
}) => {
    const classes = useStyles();
    const [hideAlert, setHideAlert] = useState(false);
    const renderIcon = type => {
        switch (type) {
            case 'custom':
                return customIcon;
            case 'error':
                return (
                    <Error id="error-icon" className="icon" aria-label="Error alert." focusable aria-hidden="false" />
                );
            case 'error_outline':
                return (
                    <ErrorOutline
                        id="error-outline-icon"
                        className="icon"
                        aria-label="Error alert."
                        focusable
                        aria-hidden="false"
                    />
                );
            case 'warning':
                return (
                    <Warning
                        id="warning-icon"
                        className="icon"
                        aria-label="Important alert."
                        focusable
                        aria-hidden="false"
                    />
                );
            case 'info':
                return <Info id="info-icon" className="icon" aria-label="Alert." focusable aria-hidden="false" />;
            case 'info_outline':
                return (
                    <InfoOutlined
                        id="info-outline-icon"
                        className="icon"
                        aria-label="Alert."
                        focusable
                        aria-hidden="false"
                    />
                );
            case 'help':
                return <Help id="help-icon" className="icon" aria-label="Help alert." focusable aria-hidden="false" />;
            case 'help_outline':
                return (
                    <HelpOutline
                        id="help-outline-icon"
                        className="icon"
                        aria-label="Help aler.t"
                        focusable
                        aria-hidden="false"
                    />
                );
            case 'done':
                return (
                    <Done id="done-icon" className="icon" aria-label="Success alert." focusable aria-hidden="false" />
                );
            default:
                return (
                    <Error id="error-icon" className="icon" aria-label="Error alert." focusable aria-hidden="false" />
                );
        }
    };
    const hideThisAlert = () => {
        setHideAlert(true);
    };

    if (!!hideAlert) {
        return null;
    } else {
        return (
            <div
                className={classNames(classes[!!customIcon ? customType : type], classes.common)}
                data-testid={alertId}
                id={alertId}
            >
                <Grid
                    container
                    spacing={1}
                    justify="center"
                    alignItems="center"
                    alignContent="center"
                    className={classes.wrapper}
                >
                    <Grid item xs={12} sm className={action && !disableAlertClick && classes.linked}>
                        <Grid container justify="center" alignItems="center" alignContent="center">
                            <Grid
                                item
                                className={`${classes.icon} alert-icon ${wiggle ? classes.wiggler : ''}`}
                                onClick={!disableAlertClick && action}
                                onKeyDown={!disableAlertClick && action}
                                id={`${alertId}-action-icon-button`}
                                data-testid={`${alertId}-action-icon-button`}
                            >
                                {showLoader ? (
                                    <CircularProgress id="spinner" className="spinner" size={38} thickness={3} />
                                ) : (
                                    renderIcon(type)
                                )}
                            </Grid>
                            <Grid
                                item
                                xs
                                className={`${classes.text} alert-text`}
                                onClick={!disableAlertClick && action}
                                onKeyDown={!disableAlertClick && action}
                                id={`${alertId}-action-message-button`}
                                data-testid={`${alertId}-action-message-button`}
                            >
                                <strong>{title}</strong>
                                &nbsp;
                                {message}
                            </Grid>
                            {allowDismiss && dismissAction && (
                                <Hidden smUp>
                                    <Grid item className={classes.dismissButton}>
                                        <IconButton
                                            onClick={dismissAction}
                                            aria-label={dismissTitle}
                                            id={`${alertId}-dismiss-button-mobile`}
                                            data-testid={`${alertId}-dismiss-button-mobile`}
                                        >
                                            <Close />
                                        </IconButton>
                                    </Grid>
                                </Hidden>
                            )}
                            {canHide && (
                                <Hidden smUp>
                                    <Grid item className={classes.dismissButton}>
                                        <IconButton
                                            onClick={hideThisAlert}
                                            aria-label={dismissTitle}
                                            id={`${alertId}-hide-button-mobile`}
                                            data-testid={`${alertId}-hide-button-mobile`}
                                        >
                                            <Close />
                                        </IconButton>
                                    </Grid>
                                </Hidden>
                            )}
                        </Grid>
                    </Grid>
                    {action && actionButtonLabel && (
                        <Grid item xs sm="auto" className={classes.actionButton}>
                            <Button
                                variant="text"
                                children={actionButtonLabel}
                                onClick={action}
                                fullWidth
                                className="action alert-button"
                                id={`${alertId}-action-button`}
                                data-testid={`${alertId}-action-button`}
                            />
                        </Grid>
                    )}
                    {allowDismiss && dismissAction && (
                        <Hidden xsDown>
                            <Grid item className={classes.dismissButton}>
                                <IconButton
                                    onClick={dismissAction}
                                    aria-label={dismissTitle}
                                    id={`${alertId}-dismiss-button`}
                                    data-testid={`${alertId}-dismiss-button`}
                                >
                                    <Close />
                                </IconButton>
                            </Grid>
                        </Hidden>
                    )}
                    {!!canHide && (
                        <Hidden xsDown>
                            <Grid item className={classes.dismissButton}>
                                <IconButton
                                    onClick={hideThisAlert}
                                    aria-label={dismissTitle}
                                    id={`${alertId}-hide-button`}
                                    data-testid={`${alertId}-hide-button`}
                                >
                                    <Close />
                                </IconButton>
                            </Grid>
                        </Hidden>
                    )}
                </Grid>
            </div>
        );
    }
};

Alert.propTypes = {
    action: PropTypes.func,
    actionButtonLabel: PropTypes.string,
    allowDismiss: PropTypes.bool,
    canHide: PropTypes.bool,
    customIcon: PropTypes.any,
    customType: PropTypes.oneOf([
        null,
        'error',
        'error_outline',
        'warning',
        'info',
        'info_outline',
        'help',
        'help_outline',
        'done',
        'custom',
    ]),
    disableAlertClick: PropTypes.bool,
    dismissAction: PropTypes.func,
    dismissTitle: PropTypes.string,
    message: PropTypes.any.isRequired,
    showLoader: PropTypes.bool,
    alertId: PropTypes.string,
    title: PropTypes.string,
    type: PropTypes.oneOf([
        'error',
        'error_outline',
        'warning',
        'info',
        'info_outline',
        'help',
        'help_outline',
        'done',
        'custom',
    ]),
    wiggle: PropTypes.bool,
};

Alert.defaultProps = {
    allowDismiss: true,
    canHide: true,
    customIcon: null,
    customType: null,
    disableAlertClick: false,
    dismissTitle: 'Click to dismiss this alert',
    message:
        'Some items on your page have not loaded properly and may not work as expected. This issue has been automatically reported.',
    showLoader: false,
    type: 'error',
    wiggle: null,
};

export default Alert;
