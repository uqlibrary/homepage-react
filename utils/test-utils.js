/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { Route } from 'react-router';
import { mui1theme } from 'config/theme';
import { Provider } from 'react-redux';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';

import preview from 'jest-preview';

import { getStore } from '../src/config/store';
import Immutable from 'immutable';
import { createMemoryHistory } from 'history';

const domTestingLib = require('@testing-library/dom');
const reactTestingLib = require('@testing-library/react');

const { configure } = domTestingLib;

configure(config => ({
    ...config,
    testIdAttribute: 'id',
}));

const AllTheProviders = props => {
    return (
        <MuiThemeProvider theme={mui1theme}>
            <MuiPickersUtilsProvider utils={MomentUtils}>{props.children}</MuiPickersUtilsProvider>
        </MuiThemeProvider>
    );
};

AllTheProviders.propTypes = {
    children: PropTypes.node,
};

export const rtlRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {},
) => {
    return render(
        <AllTheProviders>
            <Router history={history}>{ui}</Router>
        </AllTheProviders>,
    );
};

export const RenderWithRouter = ({
    children,
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
}) => (
    <AllTheProviders>
        <Router history={history}>{children}</Router>
    </AllTheProviders>
);
export const renderWithRedux = ({ initialState }) => render => {
    return {
        ...render,
        store: getStore({ initialState, history: render.history }),
    };
};

export const withRouter = ({
    route = '/',
    path = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
} = {}) => WrappedComponent => {
    return (
        <Router history={history}>
            <Route path={path} children={WrappedComponent} />
        </Router>
    );
};

export const withRedux = (initialState = Immutable.Map()) => WrappedComponent => {
    return <Provider store={getStore(initialState)}>{WrappedComponent}</Provider>;
};

// eslint-disable-next-line react/prop-types
export const WithRedux = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>{children}</Provider>
);

export const WithReduxStore = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>
        <AllTheProviders>{children}</AllTheProviders>
    </Provider>
);

module.exports = {
    ...domTestingLib,
    ...reactTestingLib,
    rtlRender,
    renderWithRouter,
    renderWithRedux,
    withRedux,
    withRouter,
    WithRedux,
    WithReduxStore,
    RenderWithRouter,
    preview,
};
