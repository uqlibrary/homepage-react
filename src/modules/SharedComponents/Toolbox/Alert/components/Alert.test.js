import React from 'react';
import { rtlRender, fireEvent } from 'test-utils';
import Alert from './Alert';
import mediaQuery from 'css-mediaquery';

function setup(testProps = {}) {
    return rtlRender(
        <Alert
            title="Title"
            message="Message"
            type="warning"
            allowDismiss
            dismissAction={jest.fn()}
            action={jest.fn()}
            actionButtonLabel="button"
            {...testProps}
        />,
    );
}

function createMatchMedia(width) {
    return query => ({
        matches: mediaQuery.match(query, { width }),
        addListener: () => {},
        removeListener: () => {},
    });
}

describe('Alert', () => {
    beforeAll(() => {
        window.matchMedia = createMatchMedia(window.innerWidth);
    });

    it('should render default view', () => {
        const { getByText, getByTestId } = setup();
        expect(getByText(/Title*/)).toBeInTheDocument();
        expect(getByText('Message')).toBeInTheDocument();
        expect(getByTestId('warning-icon')).toBeInTheDocument();
        expect(toJson(getByText)).toMatchSnapshot();
    });

    it('should render error type of alert', () => {
        const { getByText, getByTestId } = setup({
            title: 'This is an error title',
            type: 'error',
            message: 'This is an error message',
        });

        expect(getByText(/This is an error title*/)).toBeInTheDocument();
        expect(getByText('This is an error message')).toBeInTheDocument();
        expect(getByTestId('error-icon')).toBeInTheDocument();
        expect(toJson(getByText)).toMatchSnapshot();
    });

    it('should render custom alert', () => {
        const { getByText } = setup({
            title: 'This is a custom title',
            type: 'custom',
            message: 'This is a custom message',
            customIcon: 'test',
            customType: 'error',
            wiggle: true,
        });

        expect(getByText(/This is a custom title*/)).toBeInTheDocument();
        expect(getByText('This is a custom message')).toBeInTheDocument();
        expect(toJson(getByText)).toMatchSnapshot();
    });

    it('should render error outline type of alert', () => {
        const { getByText, getByTestId } = setup({
            title: 'This is an error outline title',
            type: 'error_outline',
            message: 'This is an error outline message',
        });

        expect(getByText(/This is an error outline title*/)).toBeInTheDocument();
        expect(getByText('This is an error outline message')).toBeInTheDocument();
        expect(getByTestId('error-outline-icon')).toBeInTheDocument();
        expect(toJson(getByText)).toMatchSnapshot();
    });

    it('should render info type of alert', () => {
        const { getByText, getByTestId } = setup({
            title: 'This is an info title',
            type: 'info',
            message: 'This is an info message',
        });

        expect(getByText(/This is an info title*/)).toBeInTheDocument();
        expect(getByText('This is an info message')).toBeInTheDocument();
        expect(getByTestId('info-icon')).toBeInTheDocument();
        expect(toJson(getByText)).toMatchSnapshot();
    });

    it('should render info outline type of alert', () => {
        const { getByText, getByTestId } = setup({
            title: 'This is an info outline title',
            type: 'info_outline',
            message: 'This is an info outline message',
        });

        expect(getByText(/This is an info outline title*/)).toBeInTheDocument();
        expect(getByText('This is an info outline message')).toBeInTheDocument();
        expect(getByTestId('info-outline-icon')).toBeInTheDocument();
        expect(toJson(getByText)).toMatchSnapshot();
    });

    it('should render help type of alert', () => {
        const { getByText, getByTestId } = setup({
            title: 'This is an help title',
            type: 'help',
            message: 'This is an help message',
        });

        expect(getByText(/This is an help title*/)).toBeInTheDocument();
        expect(getByText('This is an help message')).toBeInTheDocument();
        expect(getByTestId('help-icon')).toBeInTheDocument();
        expect(toJson(getByText)).toMatchSnapshot();
    });
    it('should render help outline type of alert', () => {
        const { getByText, getByTestId } = setup({
            title: 'This is an help outline title',
            type: 'help_outline',
            message: 'This is an help outline message',
        });

        expect(getByText(/This is an help outline title*/)).toBeInTheDocument();
        expect(getByText('This is an help outline message')).toBeInTheDocument();
        expect(getByTestId('help-outline-icon')).toBeInTheDocument();
        expect(toJson(getByText)).toMatchSnapshot();
    });

    it('should render done type of alert', () => {
        const { getByText, getByTestId } = setup({
            title: 'This is an done title',
            type: 'done',
            message: 'This is an done message',
        });

        expect(getByText(/This is an done title*/)).toBeInTheDocument();
        expect(getByText('This is an done message')).toBeInTheDocument();
        expect(getByTestId('done-icon')).toBeInTheDocument();
        expect(toJson(getByText)).toMatchSnapshot();
    });

    it('should render clickable action button', () => {
        const actionFn = jest.fn();
        const { getByText, getByTestId } = setup({
            title: 'This is an error title',
            type: 'error',
            message: 'This is an error message',
            action: actionFn,
            actionButtonLabel: 'Dismiss',
        });

        expect(getByText(/This is an error title*/)).toBeInTheDocument();
        expect(getByText('This is an error message')).toBeInTheDocument();
        expect(getByTestId('error-icon')).toBeInTheDocument();
        expect(getByTestId('action-button')).toBeInTheDocument();

        fireEvent.click(getByTestId('action-button'));
        expect(actionFn).toHaveBeenCalledTimes(1);
        expect(toJson(getByText)).toMatchSnapshot();
    });

    it('should dismiss alert', () => {
        const dismissfn = jest.fn();
        const { getByTestId } = setup({
            dismissAction: dismissfn,
            type: null,
        });

        fireEvent.click(getByTestId('dismiss'));
        expect(dismissfn).toHaveBeenCalledTimes(1);
        expect(toJson(getByTestId)).toMatchSnapshot();
    });

    it('should display dismiss button for smaller screen size', () => {
        window.matchMedia = createMatchMedia(256);
        const { getByTestId } = setup();
        expect(getByTestId('dismiss-mobile')).toBeInTheDocument();
        expect(toJson(getByTestId)).toMatchSnapshot();
    });

    it('should display loader', () => {
        const { getByTestId } = setup({
            showLoader: true,
        });

        expect(getByTestId('spinner')).toBeInTheDocument();
        expect(toJson(getByTestId)).toMatchSnapshot();
    });

    it('should fire an action when user clicks on the message text', () => {
        const actionFn = jest.fn();
        const { getByText } = setup({
            message: 'Click me',
            action: actionFn,
        });

        fireEvent.click(getByText(/Click me/));
        expect(actionFn).toHaveBeenCalledTimes(1);
        expect(toJson(getByText)).toMatchSnapshot();
    });

    it('should not fire an action when user clicks on the message text and no action is assigned', () => {
        const actionFn = jest.fn();
        const { getByText } = setup({
            message: 'Click me',
            action: null,
        });

        fireEvent.click(getByText(/Click me/));
        expect(actionFn).toHaveBeenCalledTimes(0);
        expect(toJson(getByText)).toMatchSnapshot();
    });
});
