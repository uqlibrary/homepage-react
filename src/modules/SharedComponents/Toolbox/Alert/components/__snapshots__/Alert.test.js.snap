// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should dismiss alert 1`] = `undefined`;

exports[`Alert should display dismiss button for smaller screen size 1`] = `undefined`;

exports[`Alert should display loader 1`] = `undefined`;

exports[`Alert should fire an action when user clicks on the message text 1`] = `undefined`;

exports[`Alert should not fire an action when user clicks on the message text and no action is assigned 1`] = `undefined`;

exports[`Alert should render clickable action button 1`] = `undefined`;

exports[`Alert should render custom alert 1`] = `undefined`;

exports[`Alert should render default view 1`] = `undefined`;

exports[`Alert should render done type of alert 1`] = `undefined`;

exports[`Alert should render error outline type of alert 1`] = `undefined`;

exports[`Alert should render error type of alert 1`] = `undefined`;

exports[`Alert should render help outline type of alert 1`] = `undefined`;

exports[`Alert should render help type of alert 1`] = `undefined`;

exports[`Alert should render info outline type of alert 1`] = `undefined`;

exports[`Alert should render info type of alert 1`] = `undefined`;
